### PR TITLE
Add missing type definitions to codebase

### PR DESCRIPTION
**Pull request by snario**
Ran `npx typesync`

### CODE REVIEW

**Review by annieke — APPROVED** [review]

**Comment by tynes** [comment]
> [Strange that the Geth unit tests fail...](https://github.com/ethereum-optimism/optimism/runs/2521042651)

That test failing is a known intermittent failure

**Comment by smartcontracts** [comment]
Needs a few conflict resolutions but LGTM otherwise

**Comment by maurelian** [comment]
This PR only affects non-production code, so should be good to merge without further review once the conflicts are fixed. 
(Per our [contributing guidelines](https://github.com/ethereum-optimism/.github/blob/master/CONTRIBUTING.md#steps-for-pr-reviewers.))

**Pull request merged**
